azure/create-spn.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+# set -x
+set -e
+
+script_name=create-spn.sh
+display_name=""
+subscription=""
+
+spn=""
+
+show_help()
+{
+  echo "Required permissions:"
+  echo "  Microsoft.Authorization/roleAssignments/write"
+  echo "Required tools:"
+  echo "  jq"
+  echo "Usage:"
+  echo "  $0 --display-name NAME [options]"
+  echo ""
+  echo "Options:"
+  echo "  -d, --display-name: The name of Azure AD App."
+  echo "  -s, --subscription: The Azure Subscription ID used to create resources."
+  echo "  -h, --help:         Show this help."
+  echo ""
+}
+
+check()
+{
+  # jq is required.
+  hash jq > /dev/null 2>&1 || { show_help; echo "Error: please install jq on the system"; }
+  check_display_name
+  check_subscription
+}
+
+check_display_name()
+{
+  if [[ "${display_name}" == "" ]]; then
+    show_help
+    echo "Error: missing -d, --display-name to specify Azure AD App name"
+    exit 1
+  fi
+}
+
+check_subscription()
+{
+  if [[ "${subscription}" == "" ]]; then
+    show_help
+    echo "Error: missing -s, --subscription to specify Azure Subscription"
+    exit 1
+  fi
+}
+
+while [[ $# -gt 0 ]]; do
+  key="$1"
+
+  case $key in
+    -h|--help)
+      show_help
+      exit 0
+      ;;
+    -d|--display-name)
+      display_name="$2"
+      check_display_name
```

Comment:
this will be called in L106 implicitly.

Reply:
This check is needed because the script will exit if the command is like `./create-spn.sh -d -s xxx`, which the deeper cause is we use `set -e` and shift error.